later/lib/database/repository/record_repository.dart

```diff
@@ -1,37 +1,60 @@
-import 'dart:async';
-
 import 'package:injectable/injectable.dart';
-import 'package:isar/isar.dart';
+import 'package:later/database/datasources/records_data_source.dart';
 import 'package:later/database/models/record_model.dart';
+import 'package:later/domain/models/record_model.dart';
+import 'package:later/domain/remap/record_remapper.dart';
+import 'package:later/domain/repository/record_repository.dart';
 
 @lazySingleton
-class RecordRepository {
-  late final Isar _isar;
-
-  // We must preinitialize it first
-  @FactoryMethod(preResolve: true)
-  static Future<RecordRepository> init() async {
-    final repo = RecordRepository()
-      .._isar = await Isar.open([RecordModelSchema]);
-
-    return repo;
-  }
-
-  Future<RecordModel> create({required String title}) async {
+class RecordRepositoryImpl implements RecordsRepository {
```

Comment:
Another thing: watch for the file name. It should be `record_repository_impl.dart`, same as the class name within it.